src/NuGet.Services.EndToEnd/Support/AssemblyMetadataPackageFile.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Services.EndToEnd.Support.Utilities;
+
+namespace NuGet.Services.EndToEnd.Support
+{
+    /// <summary>
+    /// Provides files for <see cref="PackageBuilder"/> from test assembly metadata
+    /// </summary>
+    internal class AssemblyMetadataPackageFile : IPackageFile
```

Comment:
The name of this is kind of confusing. Why isn't this just "PackageFile"?

Reply:
There is a `PhysicalPackageFile` provided by the library indicating that it represents "physical" files (i.e. the ones that a present in a file system). `AssemblyMetadataPackageFile` represents "files" that come from the test assembly metadata.